tests/queries/0_stateless/00757_enum_defaults.sql

```diff
@@ -1,2 +1,12 @@
 select os_name, count() from (SELECT CAST('iphone' AS Enum8('iphone' = 1, 'android' = 2)) AS os_name) group by os_name WITH TOTALS;
 select toNullable(os_name) AS os_name, count() from (SELECT CAST('iphone' AS Enum8('iphone' = 1, 'android' = 2)) AS os_name) group by os_name WITH TOTALS;
+
+DROP TABLE IF EXISTS auto_assgin_enum;
+DROP TABLE IF EXISTS auto_assgin_enum1;
+
+CREATE TABLE auto_assgin_enum (x enum('a', 'b')) ENGINE=MergeTree() order by x;
+CREATE TABLE auto_assgin_enum1 (x enum('a' = 1, 'b')) ENGINE=MergeTree() order by x; -- { serverError 223 }
+INSERT INTO auto_assgin_enum VALUES('a'), ('b');
+select * from auto_assgin_enum;
```

Comment:
The test does not check the assigned numbers.
You can use CAST for it.

Reply:
should we check assigned numbers? it's straight forward increment from 1, and is it important? the only expected result - it's a set of unique numbers hidden behind names

Reply:
@alexey-milovidov BTW it's an interesting question - underlying enum type is signed, however usually enums are used with positive numbers - so effectively the available range is cut in half by users practice. Besides if we want to test numbers assignment above 8 bit range - it's not above 255 - it's above 127...

Reply:
I check for mysql, it's enum value increase from 1，so if we execute "select * from auto_assign_enum where x=1;" it should return 1
```
mysql> show create table t1;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                    |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `a` int DEFAULT NULL,
  `b` enum('a','b') DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> select * from t1 where b = 1;
+------+------+
| a    | b    |
+------+------+
|    1 | a    |
+------+------+
1 row in set (0.00 sec)

mysql> select * from t1 where b = 2;
+------+------+
| a    | b    |
+------+------+
|    1 | b    |
+------+------+
1 row in set (0.00 sec)
```

Reply:
> 

I will add tests.